### PR TITLE
Feature/layouts

### PR DESCRIPTION
**Pull request by romaninsh**
We need a couple of usable full-screen layouts coming as a standard.

This pull request adds:

 - Centered - layout that puts content in the center of the page.
 - Admin - layout with main menu/dropdowns on top, user icon on the right and wide content

Also demos are not neatly organised. Simply clone and open "demos" folder in the browser.

screen shot 2017-02-07 at 10 49 21

### CODE REVIEW

File: src/View.php [hunk 1]
```diff
@@ -134,11 +134,11 @@ public function __construct($defaults = [])
...
      * @return Model
      */
-    public function setModel(\atk4\data\Model $model)
+    public function setModel(\atk4\data\Model $m)
     {
-        $this->model = $model;
+        $this->model = $m;
 
-        return $model;
+        return $m;
```
Comment (romaninsh) [inline src/View.php]: why not leave $model ?

Comment (DarkSide666) [inline src/View.php]: ok we can leave $model, but then in method comments it should be $model too :)

**Comment by romaninsh** [comment]
Dark, I have added some basic documentation on the "Layouts". It's not deep yet, but I'm not sure in which directions Layouts will go. They are a new concept for ATK so I'd like to give it some consideration.

Do verify the rest and let's switch to Grids next.

File: src/Button.php [hunk 2]
```diff
@@ -43,6 +43,26 @@ public function renderView()
...
 
     /**
+     * Makes button into a "<a>" element with a link.
+     */
+    public function link($url)
+    {
+        $this->element = 'a';
+        if (is_string($url)) {
+            $sch = substr($url, 0, 3);
+            if ($sch === 'htt' || $sch = '://') {
```
Comment (romaninsh) [inline src/Button.php]: I'm actually going to do like this: all strings are inserted as-is. Array routed through url()

File: src/Button.php [hunk 3]
```diff
@@ -43,6 +43,26 @@ public function renderView()
...
+                $this->setAttr('target', '_blank');
```
Comment (romaninsh) [inline src/Button.php]: should i drop target setting or add as argument to link() ?

Comment (DarkSide666) [inline src/Button.php]: i think - drop that line :)

File: src/Button.php (hunk 2 above)
Comment (DarkSide666) [inline src/Button.php]: i agree

**Pull request merged**